full-node/sov-prover-storage-manager/src/lib.rs

```diff
@@ -0,0 +1,411 @@
+mod dummy_storage;
+mod snapshot_manager;
+
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::marker::PhantomData;
+use std::sync::{Arc, RwLock};
+
+use sov_rollup_interface::da::{BlockHeaderTrait, DaSpec};
+use sov_rollup_interface::storage::HierarchicalStorageManager;
+use sov_schema_db::snapshot::{DbSnapshot, ReadOnlyLock, SnapshotId};
+use sov_state::MerkleProofSpec;
+
+use crate::dummy_storage::NewProverStorage;
+use crate::snapshot_manager::SnapshotManager;
+
+struct NewProverStorageManager<Da: DaSpec, S: MerkleProofSpec> {
+    // L1 forks representation
+    // Chain: prev_block -> child_blocks
+    chain_forks: HashMap<Da::SlotHash, Vec<Da::SlotHash>>,
+    // Reverse: child_block -> parent
+    blocks_to_parent: HashMap<Da::SlotHash, Da::SlotHash>,
+
+    latest_snapshot_id: SnapshotId,
+    block_hash_to_snapshot_id: HashMap<Da::SlotHash, SnapshotId>,
+    // Same reference for individual managers
+    snapshot_id_to_parent: Arc<RwLock<HashMap<SnapshotId, SnapshotId>>>,
+
+    state_snapshot_manager: Arc<RwLock<SnapshotManager>>,
+    native_snapshot_manager: Arc<RwLock<SnapshotManager>>,
```

Comment:
what is the difference between `state_snapshot_manager ` vs `native_snapshot_manager `?

Reply:
one is for `state_db` and another one is for `native_db` (accessory db)

I suppose ledger_db will be added here as well. 

Technically underlying maps are the same in shape, but values are different. This makes each type smaller and easier to reason.

Reply:
Renamed to accessory